utils/build-script-impl

```diff
@@ -2792,7 +2792,12 @@ for host in "${ALL_HOSTS[@]}"; do
                     # executing ninja directly, have it dump the commands it would
                     # run, strip Ninja's progress prefix with sed, and tell the
                     # shell to execute that.
-                    sh -e -x -c "$("${build_cmd[@]}" -n -v ${target} | sed -e 's/[^]]*] //')"
+                    # However, if we do this in a subshell in the ``sh -e -x -c`` line,
+                    # errors in the command will not stop the script as they should.
+                    echo "Generating dry run test command from: ${build_cmd[@]} -n -v ${target}"
+                    dry_run_command_output="$(${build_cmd[@]} -n -v ${target} | sed -e 's/[^]]*] //')"
+                    echo "Test command: ${dry_run_command_output}"
+                    sh -e -x -c "${dry_run_command_output}"
```

Comment:
llvm is failing to build when I have a space in a directory. I don't know how long this has been broken for. Maybe we should have jenkins build in ``/home/jenkins/work space`` to catch these errors.

This patch fixes a specific hack where we invoke `sh` in a way where the command substitution in its body fails and returns an empty string, but then the `sh` command executes and returns 0, and the script keeps going.